ranges.cpp

```diff
@@ -484,6 +507,214 @@ constexpr auto transform(Func p) {
+        friend PRS_VIEWS_CONSTEXPR bool operator==(iterator const& lhs, iterator const& rhs) noexcept {
+            bool b = true;
+
+            // This may be the most important line in the view, begin is not init until ++ operator is called
+            // but end is initialized, then we avoid begin == end
+            if (lhs.m_bInitialized.has_value() && rhs.m_bInitialized.has_value())
+            {
+                b = lhs.m_bInitialized.value() == rhs.m_bInitialized.value();
+            }
+            
+            return (lhs.m_iter == rhs.m_iter) && b;
+        }
+        friend PRS_VIEWS_CONSTEXPR bool operator!=(iterator const& lhs, iterator const& rhs) noexcept {
+            return !(lhs == rhs);
+        }
+
+        constexpr bool operator==(typename View::iterator rhs) const {
+            return m_iter == rhs;
+        }
+        constexpr bool operator!=(typename View::iterator rhs) const {
+            return m_iter != rhs;
+        }
+    };
+
+    template<typename V, typename = std::enable_if_t<!std::is_same_v<std::remove_cv_t<std::remove_reference_t<V>>, startAt_view>>>
+    constexpr startAt_view(V&& v, startAt_func_t f) noexcept : 
+        m_view{std::forward<V>(v)}, 
+        m_func{ std::move(f) },
+        m_begin{ iterator{ m_view.begin(), this } }, 
+        m_end{ iterator{ m_view.end(), this } } 
+    {
+        static_assert(detail::is_range_view_v<V>, "startAt_view can only be constructed from a range!");
+    }
+
+    constexpr startAt_view(startAt_view const& rhs)
+        : m_view{ rhs.m_view },
+        m_func{ rhs.m_func },
+        m_begin{ iterator{ m_view.begin(), this } },
+        m_end{ iterator{ m_view.end(), this } }
+    {}
+    constexpr startAt_view& operator=(startAt_view const& rhs) {
+        startAt_view copy{ rhs };
+        copy.swap(*this);
+        return *this;
+    }
+    constexpr startAt_view(startAt_view&& rhs) noexcept :
+        m_view{std::move(rhs.m_view)}, 
+        m_func{ std::move(rhs.m_func) },
+        m_begin{ iterator{ m_view.begin(), this } }, 
+        m_end{ iterator{ m_view.end(), this } } 
+    {
+    }
+    constexpr startAt_view& operator=(startAt_view&& rhs) noexcept {
+        startAt_view copy{ std::move(rhs) };
+        copy.swap(*this);
+        return *this;
+    }
+    ~startAt_view() = default;
+
+    friend constexpr void swap(startAt_view& lhs, startAt_view& rhs) noexcept {
+        using std::swap;
+        swap(lhs.m_view, rhs.m_view);
+        swap(lhs.m_func, rhs.m_func);
+        swap(lhs.m_begin, rhs.m_begin);
+        swap(lhs.m_end, rhs.m_end);
+        lhs.m_begin.m_view = std::addressof(lhs);
+        lhs.m_end.m_view = std::addressof(lhs);
+        rhs.m_begin.m_view = std::addressof(rhs);
+        rhs.m_end.m_view = std::addressof(rhs);
+    }
+
+    constexpr auto begin() {
+        if (!m_start.has_value()) {
+            _subGetLimitIter(m_start, false);
+        }
+
+        return m_start.value();
+    }
+    constexpr auto end(){
```

Comment:
I really don't understand how the "element right before `begin`" is determined here :grin: 
You're checking the predicate in both cases